gitbook-docs/data_model_multiple_values.md

```diff
@@ -26,99 +26,153 @@ It is quite possible for a key value to come from more than one device. eg posit
-        "n2k": {
-          "actisense-115-129026": {
-            "value": 102.29,
-            "bus": "/dev/actisense",
-            "timestamp": "2014-08-15-16: 00: 01.083",
-            "src": "115",
-            "pgn": "129026"
-          },
-          "actisense-201-130577": {
-            "value": 102.29,
-            "bus": "/dev/actisense",
-            "timestamp": "2014-08-15-16: 00: 00.085",
-            "src": "201",
-            "pgn": "130577"
+        "courseOverGroundMagnetic": {
+          "value": 3.615624078431453,
+          "$source": "nmea2.II",
+          "timestamp": "2017-03-04T14:58:48.000Z",
+          "sentence": "VTG",
+          "values": {
+            "nmea1.II": {
+              "value": 3.6376152270065814,
+              "sentence": "VTG",
+              "timestamp": "2017-03-04T14:58:47.000Z"
+            },
+            "nmea2.II": {
+              "value": 3.615624078431453,
+              "sentence": "VTG",
+              "timestamp": "2017-03-04T14:58:48.000Z"
+            }
           }
         }
       }
     }
+  },
+  "self": "urn:mrn:signalk:uuid:c0d79334-4e25-4245-8892-54e8ccc8021d",
+  "version": "0.1.0",
+  "sources": {
+    "nmea1": {
+      "label": "nmea1",
+      "type": "NMEA0183",
+      "II": {
+        "talker": "II",
+        "sentences": {
+          "VTG": "2017-03-04T14:58:47.000Z"
+        }
+      }
+    },
+    "nmea2": {
+      "label": "nmea2",
+      "type": "NMEA0183",
+      "II": {
+        "talker": "II",
+        "sentences": {
+          "VTG": "2017-03-04T14:58:48.000Z"
+        }
+      }
+    }
   }
 }
 ```
+###Update messages
 
-### Rules
+When a client subscribes to `navigation.courseOverGroundMagnetic`, they recieve _all_ the values held. The update message does not include the `values` path, the case above looks like:
 
-Now simple rules can apply to obtain the default, or any specific value:
 
-* The implementation must ensure that the `key.value` holds an appropriate value. This will be easy if there is only one, and will probably be user configured if more.
-* If the `source` value is `string` then it is a reference key to the source object, and can be a relative or absolute signalk key.
-* The `source` (as a reference string) also provides a mechanism to handle deprecated keys.
-* If the `source` value is a `json object` then it holds meta data on the source of the value.
-* Alternate sources must be discovered manually, or by implementation specific meta-data.
-
-To see all the entries, use the REST api or subscribe to the parent object. A given device may choose to subscribe to a specific entry in the object, allowing multiple displays of the key, or users of the various values. The 'list' verb used in a query message can provide available keys.
+```
+{
+  "context": "vessels.urn:mrn:signalk:uuid:c0d79334-4e25-4245-8892-54e8ccc8021d",
+  "updates": [
+    {
+      "source": {
+        "label": "nmea2",
+        "type": "NMEA0183",
+        "II": {
```

Comment:
This is not correct. `source` should have just talker field, `II` and `sentences` are just in the full tree.